soroban-env-host/benches/common/measure.rs

```diff
@@ -160,32 +171,63 @@ impl Measurements {
     }
 
     pub fn fit_model_to_cpu(&self) -> FPCostModel {
-        let data = self
-            .measurements
+        // data must be preprocessed
+        assert_eq!(self.measurements.len(), self.net_meas_per_iter.len());
+
+        let (x, y): (Vec<_>, Vec<_>) = self
+            .net_meas_per_iter
             .iter()
-            .map(|m| FPPoint {
-                x: m.input as f64,
-                y: m.cpu_insns as f64,
-            })
-            .collect();
-        fit_model(&data)
+            .skip(1) // for some reason first data point is often flakey, skip it
+            .map(|m| (m.inputs.unwrap_or(0), m.cpu_insns))
+            .unzip();
+
+        fit_model(x, y)
     }
 
     pub fn fit_model_to_mem(&self) -> FPCostModel {
-        let data = self
-            .measurements
+        // data must be preprocessed
+        assert_eq!(self.measurements.len(), self.net_meas_per_iter.len());
+
+        let (x, y): (Vec<_>, Vec<_>) = self
+            .net_meas_per_iter
             .iter()
-            .map(|m| FPPoint {
-                x: m.input as f64,
-                y: m.mem_bytes as f64,
-            })
-            .collect();
-        fit_model(&data)
+            .skip(1) // for some reason first data point is often flakey, skip it
+            .map(|m| (m.inputs.unwrap_or(0), m.mem_bytes))
+            .unzip();
+
+        fit_model(x, y)
     }
 }
 
-/// HostCostMeasurement is an interface to measuring the memory and CPU
+/// HostCostMeasurement (HCM) is an interface to measuring the memory and CPU
 /// costs of a CostType: usually a block of WASM bytecode or a host function.
+/// All types of costs are modeled by the linear function (see [`CostModel`]):
+///
+///     f(x) = N_x * (a + b * Option<x>)                                    [1]
```

Comment:
Hm, I think this should be `Option<b>` not `Option<x>`, right? And really `None` just means `0` in the model?

I'm also unclear on why `N_x` is something you're putting in the model rather than just tracking during measurement. The number of iterations spent _measuring_ doesn't seem to me conceptually related to the predicted cost when evaluating the model on a particular input size.